### ISSUE TITLE
@vscode/vsce has a dependecy on vulnerable versions of parse-semver and semver

### ISSUE THREAD

**Issue by ogretmenb** [microsoft/vscode-vsce#877]
There is another issue regarding semver version
https://github.com/microsoft/vscode-vsce/pull/874

but parse-semver also depends on a vulnerable version of semver
![image](https://github.com/microsoft/vscode-vsce/assets/1428781/82be4a92-e33e-4594-9c4b-87670d6d22ef)

> npm audit report
> 
> semver  <7.5.2
> Severity: moderate
> semver vulnerable to Regular Expression Denial of Service - https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
> No fix available
> node_modules/@vscode/vsce/node_modules/semver
> node_modules/parse-semver/node_modules/semver
>   @vscode/vsce  *
>   Depends on vulnerable versions of parse-semver
>   Depends on vulnerable versions of semver
>   node_modules/@vscode/vsce
>   parse-semver  *
>   Depends on vulnerable versions of semver
>   node_modules/parse-semver
> 
> 3 moderate severity vulnerabilities

**Comment by joaomoreno** [comment]
Yup, we're trying to solve it here: https://github.com/npm/node-semver/pull/564#issuecomment-1607497618
